zerver/lib/actions.py

```diff
@@ -526,23 +572,28 @@ def create_mirror_user_if_needed(realm, email, email_to_fullname):
             return get_user_profile_by_email(email)
 
 def log_message(message):
+    # type: (Message) -> None
     if not message.sending_client.name.startswith("test:"):
         log_event(message.to_log_dict())
 
 def always_push_notify(user):
+    # type: (UserProfile) -> bool
     # robinhood.io asked to get push notifications for **all** notifyable
     # messages, regardless of idle status
     return user.realm.domain in ['robinhood.io']
 
 # Helper function. Defaults here are overriden by those set in do_send_messages
 def do_send_message(message, rendered_content = None, no_log = False, stream = None, local_id = None):
+    # type: (Union[int, Message], text_type, bool, Optional[Stream], int) -> int
     return do_send_messages([{'message': message,
                               'rendered_content': rendered_content,
                               'no_log': no_log,
                               'stream': stream,
                               'local_id': local_id}])[0]
 
 def do_send_messages(messages):
+    # type: (List[Optional[Dict[str, Any]]]) -> List[int]
+    # MutableMapping gives error with mypy so we're using Dict
```

Comment:
I don't remember why I used `List` but that's what caused the problem. `List` is invariant, so I couldn't have used a subtype of `Dict`. Changing `List` to `Sequence` has made it work, since `Sequence` is covariant.
So finally I'm using `Sequence[Optional[MutableMapping[str, Any]]]`.